app/src/main/java/com/itsaky/androidide/fragments/RecentProjectsFragment.kt

```diff
@@ -66,6 +78,134 @@ class RecentProjectsFragment : BaseFragment() {
 
+	private fun openFiltersSheet() {
+		val dialog = BottomSheetDialog(requireContext())
+		val sheet = layoutInflater.inflate(R.layout.layout_project_filters_sheet, null)
+
+		dialog.setContentView(sheet)
+		setupFilters(sheet)
+
+		viewLifecycleScope.launch {
+			viewModel.filterEvents.collect { dialog.dismiss() }
+		}
+
+		dialog.show()
+	}
+
+	@OptIn(kotlinx.coroutines.FlowPreview::class)
+	private fun setupSearchBar() {
+		viewLifecycleScope.launch {
+			searchQuery
+				.debounce(300)
+				.collect { query -> viewModel.onSearchQuery(query) }
+		}
+
+		binding.layoutFilters.searchProjectEditText.addTextChangedListener { text ->
+			searchQuery.value = text?.toString().orEmpty()
+		}
+	}
+
+	private fun setupFilters(sheet: View) {
+		val sortDropdown = sheet.findViewById<MaterialAutoCompleteTextView>(R.id.sort_dropdown)
+		val sortToggleBtn = sheet.findViewById<MaterialButton>(R.id.sort_toggle_btn)
+		val applyBtn = sheet.findViewById<MaterialButton>(R.id.apply_filters_btn)
+		val clearBtn = sheet.findViewById<MaterialButton>(R.id.clear_filters_btn)
+
+		selectedCriteria = viewModel.currentSortCriteria
+		selectedAsc = viewModel.currentSortAscending
+
+		setupSortUI(sortDropdown, sortToggleBtn)
+		clearBtn.isVisible = viewModel.hasActiveFilters
+
+		sortDropdown.setOnItemClickListener { _, _, position, _ ->
+			selectedCriteria = when (position) {
+				0 -> SortCriteria.NAME
+				1 -> SortCriteria.DATE_CREATED
+				2 -> SortCriteria.DATE_MODIFIED
+				else -> SortCriteria.NAME
+			}
+
+			clearBtn.isVisible = true
+		}
+
+		sortToggleBtn.setOnClickListener {
+			toggleSortDirection(sortToggleBtn)
+			clearBtn.isVisible = true
+		}
+
+		applyBtn.setOnClickListener {
+			selectedCriteria?.let { viewModel.onSortSelected(it) }
+			viewModel.onSortDirectionChanged(selectedAsc)
+			viewModel.notifyFiltersSaved()
+		}
+
+		clearBtn.setOnClickListener {
+			selectedCriteria = null
+			selectedAsc = true
+
+			sortDropdown.text = null
+			setupSortToggle(sortToggleBtn, true)
+			binding.layoutFilters.searchProjectEditText.text?.clear()
+
+			clearBtn.isVisible = false
+			viewModel.clearFilters()
+
+			viewModel.notifyFiltersSaved()
+		}
+	}
+
+	/**
+	 * Initializes UI according to current VM state.
+	 */
+	private fun setupSortUI(
+		sortDropdown: MaterialAutoCompleteTextView,
+		sortToggleBtn: MaterialButton
+	) {
+		val labelRes = when (selectedCriteria) {
+			SortCriteria.NAME -> R.string.sort_by_name
+			SortCriteria.DATE_CREATED -> R.string.sort_by_created
+			SortCriteria.DATE_MODIFIED -> R.string.sort_by_modified
+			null -> null
+		}
+
+		if (labelRes != null) {
+			sortDropdown.setText(getString(labelRes), false)
+		} else {
+			sortDropdown.text = null
+		}
+
+		setupSortToggle(sortToggleBtn, selectedAsc)
+	}
+
+	/**
+	 * Updates the arrow icon, the background color and the text based on ascending state.
+	 */
+	private fun setupSortToggle(button: MaterialButton, asc: Boolean) {
+		val (iconRes, colorRes, textRes) = if (asc) {
+			Triple(
```

Comment:
Nitpick, instead of `Triple`, we could use a data class:


```suggestion
			data class SortToggle(@DrawableRes val iconRes:  Int)
```

This will make the logic easier to read